File: FC.Shared/Entities/ServiceResponse.cs
using FC.Interfaces.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace FC.Shared.Entities
{
    public class ServiceResponse<T>
    {
        public ServiceResponse()
        {
            this.Params = new List<string>();
        }
        public ServiceResponse(T data, HttpStatusCode statusCode, string message, Guid? token, IRepositoryState state=null)
        {
            this.Data = data;
            this.StatusCode = statusCode;
            this.Message = message;
            this.Params = new List<string>();
            this.State = state;
            this.ResponseToken = token;
        }
        public List<string> Params { get; set; }
        public IRepositoryState State { get; set; }
        public T Data { get; set; }
        public HttpStatusCode StatusCode { get; set; }
        public string Message { get; set; }
        public Guid? RequestToken { get; set; }
        public Guid? ResponseToken { get; set; }
    }
}

File: FC.WebMVC/Scripts/Shared/Models/ServiceResponse.ts
module FC.Shared.Models {
    import CM = FC.Core.CoreModel;
    import INT = FC.Shared.Interfaces;
    import MODELS = FC.Shared.Models;
    import MODULES = FC.Modules;
    export class ServiceResponse<T> {
        constructor(r: INT.IServiceResponse<T>) {
            
            this.Data = r.Data;
            this.Message = r.Message;
            this.StatusCode = r.StatusCode;
            this.Params = r.Params;
            this.State = r.State;
            this.RequestToken = r.RequestToken;
            this.ResponseToken = r.ResponseToken;

        }
        public Data: T;
        public State: FC.Shared.ViewModels.RepositoryState;
        public Message: string;
        public StatusCode: number;
        public Params: Array<string>;
        public ResponseToken: string;
        public RequestToken: string;
    }
}